### ISSUE TITLE
linter: `no-unused-vars` should relax during destructured assingment

### ISSUE THREAD

**Issue by lukeed** [oxc-project/oxc#14764]
### What version of Oxlint are you using?

1.23.0

### What does your `.oxlintrc.json` config file look like?

```jsonc
{
  "plugins": [
    "oxc",
    "import",
    "promise",
    "typescript",
    "jsx-a11y",
    "unicorn"
    // "jsdoc"
  ],
  "categories": {
    "correctness": "error",
    "suspicious": "warn",
    "perf": "warn"
  },
  "rules": {
    "no-cond-assign": "off",
    "no-import-prefix": "off",
    "no-inner-declarations": "off",
    "no-namespace": "off",
    "no-var": "off",
    "prefer-const": "off"
  },
}
```

### What happened?

with input like this, `password` should not be reporting a `no-unused-vars` error. Its definition / extraction serves a purpose.

```ts
let { password, ...profile } = user;
return Response.json(profile);
```

**Comment by howenyap** [comment]
you can configure this by setting `ignoreRestSiblings` to `true` in your config

<img width="1920" height="1054" alt="Image" src="https://github.com/user-attachments/assets/b5b9e6c8-a229-4c55-b2f8-09f5ae086931" />
